serverless/cmd/feeder/github/main.go

```diff
@@ -177,14 +177,15 @@ func feedOnce(ctx context.Context, opts *options, forkRepo *git.Repository, ghCl
 
 	glog.V(1).Infof("CP after feeding:\n%s", string(wRaw))
 
-	branchName := fmt.Sprintf("refs/heads/witness_%s", hex.EncodeToString(wCp.Hash))
+	hexHash := hex.EncodeToString(wCp.Hash)
+	branchName := fmt.Sprintf("refs/heads/witness_%s", hexHash)
 	deleteBranch, err := gitCreateLocalBranch(forkRepo, headRef, branchName)
 	if err != nil {
 		return fmt.Errorf("failed to create git branch for PR: %v", err)
 	}
 	defer deleteBranch()
 
-	outputPath := filepath.Join(opts.logPath, "checkpoint.witnessed")
+	outputPath := filepath.Join(opts.logPath, "witness", fmt.Sprintf("checkpoint_%s", hexHash))
```

Comment:
Won't this cause conflicts when merged into `master`? If I'm reading this right, all witnesses seeing the same checkpoint will create their own cosigned version called `witness/checkpoint_${LOG_ROOT_HASH}` on their own named branch. But `witness/checkpoint_${LOG_ROOT_HASH}` will be the same for all checkpoints, so when merged to master there will be a conflict. If numRequiredSigs > 1 then we'll never get the right number?

I'm guessing that I've missed some logic elsewhere that renames this file when merging to master, but I'll leave this comment and then find the other part of the machine that does that...

Reply:
Yeah, with my noddy set up I don't hit that because the combine-on-master action tidies everything up fairly quickly, but in cases where more than 1 cosign was needed to promote you could certainly have collisions.

Just added a commit which tries to make the ID stable for a given CP+set-of-cosigs (I wanted a stable ID to avoid the feeder accidentally creating new PRs and branches all over the place)